### ISSUE TITLE
Could anyone rewrite the bert model code and document ?

### ISSUE THREAD

**Issue by DachuanZhao** [tensorflow/models#8801]
Same issue in [https://github.com/tensorflow/hub/issues/635](https://github.com/tensorflow/hub/issues/635)

## 1. The entire URL of the documentation with the issue

https://github.com/tensorflow/models/blob/master/official/nlp/bert/run_classifier.py

## 2. Describe the issue

I find a pre-trained bert in tf-hub [https://tfhub.dev/tensorflow/bert_zh_L-12_H-768_A-12/2](https://tfhub.dev/tensorflow/bert_zh_L-12_H-768_A-12/2)  and its document says that I can find how to use it in [https://github.com/tensorflow/models/blob/master/official/nlp/bert/run_classifier.py](https://github.com/tensorflow/models/blob/master/official/nlp/bert/run_classifier.py)

But in ```run_classifier.py``` , the code is ugly and difficult to understand . Since the tf document contains a transformer model example use ```tf.keras.Model``` [https://www.tensorflow.org/tutorials/text/transformer](https://www.tensorflow.org/tutorials/text/transformer) , could anyone rewrite the bert model code and document like that ?

**Comment by saberkun** [comment]
@DachuanZhao 
Here is the better documentation (colab) that probably you are looking for.
https://github.com/tensorflow/models/blob/master/official/colab/fine_tuning_bert.ipynb
It contains the clear TFHub usage.

The tutorial only implemented the single GPU case where the run_classifier.py implements multi-gpu/tpu for glue datasets with predictions. 
Thank you for suggestion. We are working on a revised version with more consolidation and explicit documentation.

**Comment by DachuanZhao** [comment]
@saberkun 
Thank you . I find it in [https://www.tensorflow.org/official_models/fine_tuning_bert](https://www.tensorflow.org/official_models/fine_tuning_bert) , too .
But It uses ```tf-nightly``` which means it's not stable . Will it be stable when the ```tf2.3``` is released ? 
By the way , the document doesn't use ```tf.keras.Model``` to give an example . I think it's more convenient for user to use bert like this : 
```
bert_layer = hub.KerasLayer(bert_url)
model = tf.keras.Sequential()
model.add(bert_layer)
model.add(other_layer)
model.summary()
```
Can it be achieved ？

**Comment by saberkun** [comment]
The TF-hub code path does not use special thing in TF2.3, so TF 2.2 should be good.
The bert model usually takes 3 inputs to handle mask and segments which are important for GLUE and SQUAD.
I am not sure how convenient keras sequential API can help here.
The example is using keras functional API. The bert encoder outputs sequence embeddings [batch size, sequence length, hidden size]. There are many ways to use this output. So the other layer could have variants too. For example, for classification task, we can use: https://github.com/tensorflow/models/blob/master/official/nlp/modeling/layers/cls_head.py#L27
We built some out-of-box keras model objects as: https://github.com/tensorflow/models/tree/master/official/nlp/modeling/models. You can pick one of them to pass in the hub.KerasLayer.

We use tf-nightly most because the master branch targets tf-nightly as we use this repo internally.
As far as I know, we have some symbols only in TF 2.3 or nightly right now.

**Comment by DachuanZhao** [comment]
@saberkun 
When I ```pip install tf-models-nightly``` , ```tensorflow``` is updated to version ```2.4.0-dev20200708``` .  (￣△￣；)
By the way , Is this bert model stable enough to use in a product envirenment ?  Or it's just for test ?
Will it be added into stable ```tensorflow``` release ?

**Comment by saberkun** [comment]
@DachuanZhao Gotcha. Good point.
We have stable releases starting each TF release as: https://pypi.org/project/tf-models-official/
pip install tf-models-official right now should use TF 2.2
To my knowledge, the related code has not change since then, so should be stable enable for product.
Especially the TF-Hub module is a stable component.

Thanks for feedback.

**Comment by DachuanZhao** [comment]
@saberkun 
God job ~! Now you can change the code in [https://www.tensorflow.org/official_models/fine_tuning_bert](https://www.tensorflow.org/official_models/fine_tuning_bert ) 
```
!pip install -q tf-nightly
!pip install -q tf-models-nightly
```
into
```
!pip install -q tensorflow
!pip install -q tf-models-official
```
By the way , I suggest that it's better to use module ```tf-nightly tf-models-nightly``` in document as less as possible . It will not be used for either product or research , so it is useless for people who use ```tensorflow``` . 
Thanks for your response ~!
